File: Backend/vs/src/main/java/com/foodapp/backend/service/AuthService.java
package com.foodapp.backend.service;

import com.foodapp.backend.dto.auth.AuthResponse;
import com.foodapp.backend.dto.auth.LoginRequest;
import com.foodapp.backend.dto.auth.RegisterRequest;
import com.foodapp.backend.dto.auth.UserProfileResponse;

public interface AuthService {
    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    UserProfileResponse getMyProfile();
}

File: Frontend/food-ordering-system/src/services/authService.js
import axiosInstance from './axiosInstance';

const authService = {
  register: (data) => axiosInstance.post('/auth/register', data),
  login: (data) => axiosInstance.post('/auth/login', data),
  getProfile: () => axiosInstance.get('/auth/me'),
};

export default authService;
